### PR TITLE
Add Topographic basemap for Land & water layer

### LINKED ISSUE #47: Switch to Topographic Basemap when Land & water is selected
## Overview

Currently, there are two basemap tiles: Default and Satellite, and there is a Land & water layer button. Since we cannot use vector tile layers without a paid subscription, and we're trying to do things easily without third party subscriptions attached in this prototype, let's switch to using the [ESRI Topographic Basemap](https://www.arcgis.com/apps/mapviewer/index.html?webmap=d5e02a0c1f2b4ec399823fdd3c2fdebd) when Land & water layer is enabled.

- When the Satellite tile is selected, the basemap should always be Satellite.
- When the Default tile is selected,
  - If Land & water is enabled, the basemap should be Topographic
  - If Land & water is disabled, the basemap should be Grayscale

### PR DESCRIPTION
**Pull request by rajadain**
## Overview

Since we can't add individual vector tiles, we switch the basemap wholesale. ~~When the Land & water layer is enabled, and the default basemap is set to active, we show the Topographic layer. Otherwise we show the Light Grey Canvas layer. The Satellite tile shows that layer regardless of Land & water state.~~

Adds a new Land & water basemap tile. Removes the Street and Land & water data layers. Updates styling for polygon and Municipal Boundary layers.

Further styling updates to be done in #49 

Closes #47 

### Demo

~~https://user-images.githubusercontent.com/1430060/187044736-63a2ea79-6927-428d-8d48-d1ee0e6d9c25.mp4~~

https://user-images.githubusercontent.com/1430060/187742964-3205b9bf-20d2-4086-99a1-7965789ec1c9.mp4

## Testing Instructions

- Checkout this branch and `server`
- Go to http://localhost:4545/draw
- [x] Ensure you no longer see Streets and Land & water data layers
- [x] Ensure you see a Land & water basemap, and switching basemaps works as expected
- [x] Ensure the styling of the sidebar matches the layout here: https://www.figma.com/file/tJPZ0lkT5C3SAyS2g2R1aA/IoW---Boundary-Sync-Tool?node-id=498%3A14698

## Checklist

- [x] `fixup!` commits have been squashed
- [x] `CHANGELOG.md` updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines
- [x] `README.md` updated if necessary to reflect the changes
- [x] CI passes after rebase

### CODE REVIEW

**Comment by rajadain** [comment]
Let's hold off on reviewing this until @gatesgodin's proposed changes are in

**Comment by caseycesari** [comment]
One quick things that are not directly related to this PR, but that could be addressed here just for efficiency's sake: can the HTML app name be switched from React App to Boundary Sync? 
image

**Comment by gatesgodin** [comment]
@rajadain I have updated the layout and styles for the tool. Please see the links below:

**Basemap tiles:** https://drive.google.com/drive/folders/1-YcXXeVhoOa_ZIi7FcCFGL8YOU88OQYC?usp=sharing
_Updated the tiles to all be in the same area. One for each basemap style._

**Basemap layout:** https://www.figma.com/file/tJPZ0lkT5C3SAyS2g2R1aA/IoW---Boundary-Sync-Tool?node-id=498%3A14698
_I removed the unneeded layers and added the third basemap._

**Layer styles:** https://www.figma.com/file/tJPZ0lkT5C3SAyS2g2R1aA/IoW---Boundary-Sync-Tool?node-id=498%3A14829
_Style updates to the municipal boundary, polygon edge, polygon face, and vertices._

**Comment by rajadain** [comment]
This is ready for review. Please see updated demo and testing instructions above.

@gatesgodin I don't think there's a way in Leaflet to do the kind of 2 color alternating stroke you defined here:

image

It's possible that with a different way of map layer organization #52 we'll be able to target it via CSS and add some styling that way. It's also possible that we may be able to just switch to a lighter stroke color when the basemap is satellite. Such work is deferred until #49.

The other design items have been implemented here.

File: src/app/src/App.css
Comment (rajadain) [inline src/app/src/App.css:5]: These are defined as `8px` and `6px` in Figma https://www.figma.com/file/tJPZ0lkT5C3SAyS2g2R1aA/IoW---Boundary-Sync-Tool?node-id=498%3A14829, but we use larger sizes here because the box sizing model is different.

With the `border-box` sizing model, borders grow inwards. I tried using `content-box` but that messed up the layout. Going with slightly larger sizes seemed like a good approximation.

**Comment by rajadain** [comment]
Added a slight transparent fill to the Municipal Boundaries as discussed in UX call [3e813d3](https://github.com/azavea/iow-boundary-tool/pull/48/commits/3e813d37f822580e102ae52ffa13ecf76afca111):

image

**Comment by rajadain** [comment]
I'm seeing an odd behavior on this PR where the map flickers on every operation:

https://user-images.githubusercontent.com/1430060/187774908-ddb7ab06-d1ac-4f94-9e66-09fa5b44660f.mp4

Going to see if I can fix that.

**Comment by rajadain** [comment]
Actually the above buggy behavior is seen on `develop` too. Going to make a new bug card for that.

**Comment by rajadain** [comment]
Alright, this is ready for a proper look.

**Comment by rajadain** [comment]
Thanks to @mstone121 for helping debug #53. It was not on `develop`, but caused by changes on this branch indeed. Fixed in [dde7089](https://github.com/azavea/iow-boundary-tool/pull/48/commits/dde708990011e41f0f4e0fadec0193f142554eb7).

**Review by mstone121 — APPROVED** [review]
Looks good. There's one `fixup` commit that needs to be squashed.

There municipal boundaries layer looks way better with a transparent fill. Especially when zoomed out.

**Comment by rajadain** [comment]
Thanks for the review! Fixups squashed, will merge once green.

**Pull request merged**
